File: app/src/api/product-catalog/data-transfer-objects/pad-series.ts
import { Brand } from './brand';
import { PadMaterial } from './pad-material';
import { PadTexture } from './pad-texture';
import { PolisherType } from './polisher-type';
import { Pad } from './pad';
import { PadSize } from './pad-size';

export class PadSeries {
    constructor(
        public id: string,
        public name: string,
        public brand: Brand,
        public polisherTypes: PolisherType[],
        public sizes: PadSize[] = [],
        public pads: Pad[] = []
    ) {}
}

File: api/src/Domain/Admin/Entities/PadSeries.cs
using System;
using System.Collections.Generic;
using DetailingArsenal.Domain.Shared;

namespace DetailingArsenal.Domain.Admin.ProductCatalog {
    public class PadSeries : Aggregate<PadSeries> {
        public string Name { get; set; }
        public Guid BrandId { get; set; }

        /// <summary>
        /// Recommended for the following polisher types.
        /// </summary>
        public List<PolisherType> PolisherTypes { get; set; }
        public List<PadSize> Sizes { get; set; }
        public List<Pad> Pads { get; set; }

        public PadSeries(Guid id, string name, Guid brandId, List<PolisherType>? polisherTypes = null, List<PadSize>? sizes = null, List<Pad>? colors = null) {
            Id = id;
            Name = name;
            BrandId = brandId;
            PolisherTypes = polisherTypes ?? new();
            Sizes = sizes ?? new();
            Pads = colors ?? new();
        }

        public PadSeries(string name, Guid brandId, List<PolisherType>? polisherTypes = null, List<PadSize>? sizes = null, List<Pad>? colors = null) {
            Id = Guid.NewGuid();
            Name = name;
            BrandId = brandId;
            PolisherTypes = polisherTypes ?? new();
            Sizes = sizes ?? new();
            Pads = colors ?? new();
        }
    }
}